### PR TITLE
GUI rewrite and /ah search command

### PR DESCRIPTION
**Pull request by szumielxd**
Just some changes for better optimization and future development:
- Replaced colllidable inventory title check with custom InventoryHolder. It should be faster and safer.
- Added command to filter items by name (`/ah search <phrase>`)
- Moved Methods#updateAuction file saving task to new thread. It should eliminate threadlocks on /ah command

### CODE REVIEW

File: src/main/java/me/badbones69/crazyauctions/api/enums/GuiType.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public enum GuiType {
+	
+	
+	BIDDING(27, "Settings.Bidding-On-Item"),
+	BUYING(9, "Settings.Buying-Item"),
+	EXPIRED_LIST(54, "Settings.Cancelled/Expired-Items", true),
+	CURRENT_LIST(54, "Settings.Players-Current-Items"),
+	CATEGORIES(54, "Settings.Categories"),
+	SHOP(54, "Settings.GUIName", true),
+	SEARCH(54, "Settings.Search", true),
```
Comment (NoJokeFNA) [inline src/main/java/me/badbones69/crazyauctions/api/enums/GuiType.java:12]: Remove the last "," as it is useless

File: src/main/java/me/badbones69/crazyauctions/api/enums/GuiType.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	;
+	
+	
+	private final int size;
+	private final String title;
+	private final boolean paginable;
+
```
Comment (NoJokeFNA) [inline src/main/java/me/badbones69/crazyauctions/api/enums/GuiType.java:19]: Why sooo many empty lines tho?

File: src/main/java/me/badbones69/crazyauctions/Methods.java [hunk 3]
```diff
@@ -439,7 +439,9 @@ public static void updateAuction() {
                 }
             }
         }
-        if (shouldSave) Files.DATA.saveFile();
+        if (shouldSave) {
+        	new Thread(Files.DATA::saveFile, "Crazy-Auctions-Save-Task").start(); // save file in new thread
```
Comment (NoJokeFNA) [inline src/main/java/me/badbones69/crazyauctions/Methods.java:443]:
```suggestion
            new Thread(Files.DATA::saveFile, "Crazy-Auctions-Save-Task").start(); // save file in new thread
```
Code-Style..

**Review by NoJokeFNA — CHANGES_REQUESTED** [review]

**Comment by RyderBelserion** [comment]
Closed pending re-write

**Pull request closed without merging**